code/modules/mob/living/simple_animal/hostile/true_changeling.dm

```diff
@@ -0,0 +1,193 @@
+	var/time_spent_as_true = 0
+	var/playstyle_string = "<b><font size=3 color='red'>We have entered our true form!</font> We are unbelievably powerful, and regenerate life at a steady rate. However, most of \
+	our abilities are useless in this form, and we must utilise the abilities that we have gained as a result of our transformation. Taking too much damage will also turn us back into a \
+	human in addition to knocking us out for a long time. Finally, we will uncontrollably revert into a human after some time due to our inability to maintain this form.</b>"
+	var/mob/living/carbon/human/stored_changeling = null //The changeling that transformed
+	var/devouring = FALSE //If the true changeling is currently devouring a human
+	var/wallcrawl = FALSE //If the true changeling is crawling around the place, allowing it to counteract gravity loss
+
+/mob/living/simple_animal/hostile/true_changeling/New()
+	..()
+	if(prob(33))
+		icon_state = "horror2"
+	else if(prob(33))
+		icon_state = "horror3"
+	spawn(0)
+		to_chat(usr, playstyle_string)
+
+/mob/living/simple_animal/hostile/true_changeling/Life()
+	..()
+	adjustBruteLoss(-TRUE_CHANGELING_PASSIVE_HEAL) //True changelings slowly regenerate
+	time_spent_as_true++ //Used for re-forming
+	if(stored_changeling && time_spent_as_true >= TRUE_CHANGELING_FORCED_REFORM)
+		death() //After a while, the ling'll revert back without being able to control it
+
+/mob/living/simple_animal/hostile/true_changeling/Stat()
+	..()
+	if(statpanel("Status"))
+		if(stored_changeling)
+			var/time_left = TRUE_CHANGELING_FORCED_REFORM - time_spent_as_true
+			time_left = Clamp(time_left, 0, INFINITY)
+			stat(null, "Time Remaining: [time_left]")
+		stat(null, "Ignoring Gravity: [wallcrawl ? "YES" : "NO"]")
+
+/mob/living/simple_animal/hostile/true_changeling/death()
+	..(1)
+	if(stored_changeling && mind)
+		visible_message("<span class='warning'>[src] lets out a furious scream as it shrinks into its human form.</span>", \
+						"<span class='userdanger'>We lack the power to maintain this form! We helplessly turn back into a human...</span>")
+		stored_changeling.loc = get_turf(src)
+		mind.transfer_to(stored_changeling)
+		stored_changeling.Unconscious(30) //Make them helpless for 30 seconds
+		stored_changeling.status_flags &= ~GODMODE
+		qdel(src)
+	else
+		visible_message("<span class='warning'>[src] lets out a waning scream as it falls, twitching, to the floor.</span>", \
+						"<span class='userdanger'>We have fallen! We begin the revival process... (this will take around 45 seconds.)</span>")
+		spawn(450)
+			if(src)
+				visible_message("<span class='warning'>[src] suddenly rises from the dead!</span>")
+				revive() //Changelings can self-revive, and true changelings are no exception
+
+/mob/living/simple_animal/hostile/true_changeling/mob_negates_gravity()
+	return wallcrawl
+
+/mob/living/simple_animal/hostile/true_changeling/adjustFireLoss(amount)
```

Comment:
We have a var for this on hostile mobs already

Reply:
what is it?

Reply:
damage_coeff

Reply:
i'm going to make thinglings howl when they get damaged by burn so i need this to exist